### PR TITLE
Reactive filtering of subject component

### PR DESCRIPTION
**Pull request by blootsvoets**
Description:
- Make subject filtering reactive
- Refactor SubjectComponent to delegate some tasks to subcomponents
- Include filters in URL.

### CODE REVIEW

File: src/main/webapp/app/shared/util/reactive-filter.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    this._value$.next(null);
+    this._error$.next('');
+    this.trigger$.next(null);
+  }
+
+  complete() {
+    this._value$.complete();
+    this._error$.complete();
+    this.trigger$.complete();
+  }
```
Comment (peyman-mohtashami) [inline src/main/webapp/app/shared/util/reactive-filter.ts:77]: This method is not used. Shouldn't we call it the onDestroy method of subject.component.ts?

**Review by peyman-mohtashami — COMMENTED** [review]
LGTM
And, same as before, great effort and nice refactoring. Thank you.

File: src/main/webapp/i18n/nl/subject.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
                 "advancedFilter": "Uitgebreid filter",
                 "error": {
                     "invalidDate": "De datum is niet geldig.",
-                    "invalidRange": "De datum is niet geldig of ligt na de Van-datum."
+                    "invalidRange": "De periode is is niet geldig: Vanaf datum is na de Tot en met datum."
```
Comment (aykl) [inline src/main/webapp/i18n/nl/subject.json:29]:
```suggestion
                    "invalidRange": "De periode is niet geldig: Vanaf datum is na de Tot en met datum."
```

File: src/main/webapp/app/shared/util/reactive-filter.ts (hunk 1 above)
Comment (blootsvoets) [reply src/main/webapp/app/shared/util/reactive-filter.ts:77]: My editor says the same, but it is called in SubjectComponent line 162:
```
        for (let filtersKey in this.filters) {
             this.filters[filtersKey].complete();
         }
```

Comment (peyman-mohtashami) [reply src/main/webapp/app/shared/util/reactive-filter.ts:77]: Yes, it's my mistake.

**Pull request merged**